static/app/views/settings/organizationAuditLog/index.tsx

```diff
@@ -84,6 +85,10 @@ function OrganizationAuditLog({organization}: Props) {
       ...prevState,
       eventType: value,
     }));
+    browserHistory.push({
+      pathname: location.pathname,
+      search: `?event=${value}`,
```

Comment:
I think you also need to initialize the state based on the location state. Without that you could load the incorrect results after a page reload.